Automattic/wp-calypso
# Jetpack Offer Reset: Refactor Calls to Translate Out of Constants File

#### Changes proposed in this Pull Request

This is a followup to https://github.com/Automattic/wp-calypso/pull/45727, which was made to quickly fix an issue where usage of translated strings was not reactive. Although that solution worked, it is not ideal in the long term.

This PR re-works the Offer Reset copy that was being passed to `translate()` in the constants file so that they are instead retrieved through a function in `utils` or `translated-copy` that requires passing `translate`. Components that use this method will ideally use the `useTranslate()` hook to pass `translate`, and thus retrieve the copy in a reactive way.

#### Testing instructions



1. Checkout the branch and start up Calypso and Jetpack Cloud.
2. In Calypso, visit the plans page. Verify that everything looks the same as it does on the production version of the plans page. Be sure to check the flows that include the details page and the upsell page.
3. Visit the plans page in another language and verify that every string on the page is translated.
4. In Jetpack Cloud, visit the pricing page. Verify that everything looks the same as it does on the production version of the pricing page.
5. Visit a localized version such as http://cloud.localhost:3001/es/pricing and verify that every string on the page is translated.

Fixes #1169247016322522-as-1194900844047779
